compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs

```diff
@@ -410,28 +410,55 @@ impl<'f> PerFunctionContext<'f> {
         &mut self,
         block_id: BasicBlockId,
         references: &mut Block,
-        instruction: InstructionId,
+        instruction_id: InstructionId,
     ) {
         // If the instruction was simplified and optimized out of the program we shouldn't analyze it.
         // Analyzing it could make tracking aliases less accurate if it is e.g. an ArrayGet
         // call that used to hold references but has since been optimized out to a known result.
         // However, if we don't analyze it, then it may be a MakeArray replacing an ArraySet containing references,
         // and we need to mark those references as used to keep their stores alive.
-        let (instruction, simplified) = {
-            let (ins, loc) = self.inserter.map_instruction(instruction);
-            match self.inserter.push_instruction_value(ins, instruction, block_id, loc) {
-                InsertInstructionResult::Results(id, _) => (id, false),
+        let (instruction, loc) = self.inserter.map_instruction(instruction_id);
+
+        let (instructions, simplified) = {
+            match self.inserter.push_instruction_value(instruction, instruction_id, block_id, loc) {
+                InsertInstructionResult::Results(id, _) => (vec![id], false),
                 InsertInstructionResult::SimplifiedTo(value) => {
                     let value = &self.inserter.function.dfg[value];
-                    let Value::Instruction { instruction, .. } = value else {
+                    if let Value::Instruction { instruction, .. } = value {
+                        (vec![*instruction], true)
```

Comment:
Nit: it'd be nice if we could avoid allocating a vec for the common case of handling 1 instruction